src/context/TodosContext.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from '../types/Todo';
+import * as api from '../api/todos';
+import { Context } from '../types/Context';
+import { Status } from '../types/Status';
+
+export const TodosContext = React.createContext<Context>({
+  todos: [],
+  setTodos: () => { },
+  errorMessage: '',
+  setErrorMessage: () => { },
+  filterTodos: Status.all,
+  setFilterTodos: () => { },
+});
+
+export const TodoUpdateContext = React.createContext({
+  // addTodo: (todo: Todo) => {},
+  // deleteTodo: (todoId: number) => {},
+  // updateTodo: (todo: Todo) => {},
+});
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterTodos, setFilterTodos] = useState<Status>(Status.all);
+
+  const USER_ID = 91;
+
+  function loadTodos() {
+    api.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }
+
+  useEffect(() => {
+    const errorDelay = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    return () => clearTimeout(errorDelay);
+  }, [errorMessage]);
```

Comment:
create an `ErrorNotification` component with all error jsx and put this logic with setTimeout there

Reply:
yup, did that in next task